### PR TITLE
PLT-1360 - Configure Pronto/Simplecov code coverage in rlovelett-ffmpeg

### PR DESCRIPTION
**Pull request by spaetzel**
First implementation of [Pronto](https://github.com/prontolabs/pronto?tab=readme-ov-file) code coverage, adding to `rlovelett-ffmpeg` first since it's open source and could be cloned outside the org for quick iteration. 

Action now adds inline PR comments for both Code coverage and Rubocop problems. 

Screenshot 2024-02-08 at 9 28 21 PM

## Github Copilot Description

This pull request primarily focuses on enhancing the continuous integration (CI) pipeline and improving code quality checks. The changes include updating the GitHub Actions workflow to trigger on both push and pull request events, adding permissions to the GitHub Actions workflow, updating the actions used in the workflow, and adding new dependencies to the Gemfile. Additionally, the SimpleCov and Undercover gems have been added for better code coverage tracking.

GitHub Actions workflow updates:

* [`.github/workflows/ci.yaml`](diffhunk://#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddL2-R48): The workflow now triggers on both push and pull request events, and additional permissions have been added for the GitHub Actions workflow. The actions used in the workflow have been updated to their latest versions, and additional dependencies have been added to the installation step. A new step has been added to setup and run Pronto for code quality checks during pull request events.

Dependency additions:

* [`Gemfile`](diffhunk://#diff-d09ea66f8227784ff4393d88a19836f321c915ae10031d16c93d67e6283ab55fR7-R22): New dependencies have been added to the Gemfile, including the RuboCop Performance and RuboCop Rake gems for enhanced code quality checks, the Faraday Retry gem for retrying network requests, and the Pronto and Pronto RuboCop gems for running code quality checks. The Undercover gem has been added for tracking untested code sections, and the Rugged, SimpleCov, and SimpleCov-Lcov gems have been added to the test group for better code coverage tracking.

Code coverage tracking:

* [`spec/spec_helper.rb`](diffhunk://#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94R6-R22): The SimpleCov and SimpleCov-Lcov gems have been required and configured for code coverage tracking. The Undercover gem has also been required for tracking untested code sections.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,28 +1,51 @@
...
   lint:
     runs-on: ubuntu-latest
     container: public.ecr.aws/docker/library/ruby:3.2.2
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Install Deps
         run: |
+          apt update && apt install cmake -y
```
Comment (Kyrluckechuck) [inline .github/workflows/ci.yaml:46]: Do we need to install `cmake` for rubocop to run? 🤔

File: Gemfile.lock [hunk 2]
```diff
@@ -25,14 +75,63 @@ GEM
...
+      imagen (>= 0.1.8)
+      rainbow (>= 2.1, < 4.0)
+      rugged (>= 0.27, < 1.7)
+    unicode-display_width (2.5.0)
+    uri (0.13.0)
 
 PLATFORMS
   ruby
 
 DEPENDENCIES
```
Comment (Kyrluckechuck) [inline Gemfile.lock:121]: Oof, that's a lot of new deps just for code coverage 😰

**Review by Kyrluckechuck — COMMENTED** [review]
Not requesting any changes for now, but want to see the output / result before approving! 😛

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (spaetzel) [reply .github/workflows/ci.yaml:46]: `cmake` is for rugged - which is used by Pronto/Undercover to perform git operations

**Comment by spaetzel** [comment]
Need to turn on read & write permissions for actions in the repo. Asked @masterful to flip this on

**Review by todd-m-kemp — DISMISSED** [review]

**Review by masterful — APPROVED** [review]

**Pull request closed without merging**